common/scala/src/whisk/core/entity/WhiskTrigger.scala

```diff
@@ -37,6 +38,44 @@ case class WhiskTriggerPut(
     annotations: Option[Parameters] = None)
 
 /**
+ * Representation of a rule to be stored inside a trigger. Contains all
+ * information needed to be able to determine if and which action is to
+ * be fired.
+ *
+ * @param action the action to be fired
+ * @param status status of the rule
+ */
+case class ReducedRule(
+    action: Namespace,
+    status: Status)
+
+/**
+ * Trigger as it is returned by the controller. Basically the same as WhiskTrigger,
+ * but not including the rule.
+ *
+ * @param namespace the namespace for the trigger
+ * @param name the name of the trigger
+ * @param parameters the set of parameters to bind to the trigger environment
+ * @param limits the limits to impose on the trigger
+ * @param version the semantic version
+ * @param publish true to share the action or false otherwise
+ * @param annotation the set of annotations to attribute to the trigger
+ * @throws IllegalArgumentException if any argument is undefined
+ */
+@throws[IllegalArgumentException]
+case class WhiskTriggerResponse(
```

Comment:
this type is borderline unnecessary - the controller can just serialize the WhiskTrigger and then subtract the rule property.